Content.Server/Chemistry/EntitySystems/SolutionContainerSystem.cs

```diff
@@ -96,6 +101,15 @@ private void InitSolution(EntityUid uid, SolutionContainerManagerComponent compo
                 : "shared-solution-container-component-on-examine-worded-amount-multiple-reagents")),
             ("desc", primary.LocalizedPhysicalDescription)));
 
+        var scannerOn = false;
+        if (_inventorySystem.TryGetSlotEntity(args.Examiner, "eyes", out var ent))
+        {
+            if (HasComp<ReagentScannerComponent>(ent))
+            {
+                scannerOn = true;
+            }
+        }
```

Comment:
Raise an event directed at the entity, instead of checking for a specific component in a specific slot.